dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/AzureOpenAIImageGeneration.cs

```diff
@@ -0,0 +1,153 @@
+    /// </summary>
+    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="logger">Application logger</param>
+    /// <param name="maxRetryCount">Maximum number of retries for HTTP requests.</param>
+    /// <param name="defaultRetryAfter">Default retry wait time in seconds for HTTP requests.</param>
+    public AzureOpenAIImageGeneration(string endpoint, string apiKey, HttpClient? httpClient = null, ILogger? logger = null, int maxRetryCount = 5, int defaultRetryAfter = 6) : base(httpClient, logger)
+    {
+        this._endpoint = $"{endpoint}openai/images/generations:submit?api-version=2023-06-01-preview";
+        this._apiKey = apiKey;
+        this._maxRetryCount = maxRetryCount;
+        this._defaultRetryAfter = defaultRetryAfter;
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> GenerateImageAsync(string description, int width, int height, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(description);
+        if (width != height || (width != 256 && width != 512 && width != 1024))
+        {
+            throw new ArgumentOutOfRangeException(nameof(width), width, "OpenAI can generate only square images of size 256x256, 512x512, or 1024x1024.");
+        }
+
+        var requestBody = Json.Serialize(new ImageGenerationRequest
+        {
+            Prompt = description,
+            Size = $"{width}x{height}",
+            Count = 1
+        });
+
+        var result = await this.GetImageGenerationAsync(this._endpoint, requestBody, cancellationToken).ConfigureAwait(false);
+        if (result.Result == null)
+        {
+            throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "Response not contains result");
+        }
+
+        return result.Result.Images[0].Url;
+    }
+
+    private async Task<AzureImageGenerationResponse> GetImageGenerationAsync(string url, string? requestBody, CancellationToken cancellationToken = default)
+    {
+        HttpResponseMessage? response = null;
+        var retryCount = 0;
+        try
+        {
+            using (var content = new StringContent(requestBody, Encoding.UTF8, "application/json"))
+            {
+                response = await this.ExecuteRequestAsync(url, HttpMethod.Post, content, cancellationToken).ConfigureAwait(false);
+            }
+
+            var responseJson = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            var result = this.JsonDeserialize<AzureImageGenerationResponse>(responseJson);
+
+            if (!response.Headers.TryGetValues("operation-location", out var locationValues) || string.IsNullOrWhiteSpace(locationValues.FirstOrDefault()))
```

Comment:
What's return in the operation-location header? Could you please put a comment with a link to a documentation describing the header, assuming it exists?

Reply:
Currently, there is no official documentation that provides detailed instructions on how to use this API. 
Based on the call example provided by Azure OpenAI Studio, we can see that the operation-location points to the query address for the generated result, which is used to obtain the final image generation result.

Reply:
That's great! I actually need clear documentation right now. I'll give the API a try and make adjustments to the implementation method.